Lib/test/test_codecs.py

```diff
@@ -27,6 +27,39 @@ def check(input, expect):
         self.assertEqual(coder(input), (expect, len(input)))
     return check
 
+# On small versions of Windows like Windows IoT or Windows Nano Server not all codepages are present
+def is_code_page_present(cp):
+    from ctypes import POINTER, WINFUNCTYPE, windll, WinError, Structure
+    from ctypes.wintypes import BOOL, UINT, BYTE, WCHAR, UINT, DWORD
+
+    MAX_LEADBYTES = 12  # 5 ranges, 2 bytes ea., 0 term.
+    MAX_DEFAULTCHAR = 2 # single or double byte
+    MAX_PATH = 260
+    class CPINFOEXW(ctypes.Structure):
+        _fields_ = [("MaxCharSize", UINT),
+                    ("DefaultChar", BYTE*MAX_DEFAULTCHAR),
+                    ("LeadByte", BYTE*MAX_LEADBYTES),
+                    ("UnicodeDefaultChar", WCHAR),
+                    ("CodePage", UINT),
+                    ("CodePageName", WCHAR*MAX_PATH)]
+
+    prototype = WINFUNCTYPE(BOOL, UINT, DWORD, POINTER(CPINFOEXW))
+    GetCPInfoEx = prototype(("GetCPInfoExW", windll.kernel32))
+
+    def errcheck(result, func, args):
+        if not result:
+            from ctypes import WinError
+            raise OSError(2, "code page not found")
+        return args
+
+    from ctypes import WinError
+    try:
+        GetCPInfoEx.errcheck = errcheck
```

Comment:
Can't we just check the return value? (Or return it directly, since it seems to be the bool that we're returning from this function anyway)

Reply:
In fact, can we check `ctypes.get_last_error()` for a specific error code that means "missing"? (This is Windows GetLastError() but properly preserved across other API calls)

Reply:
I simplified this to just return it directly.  I tested that it works on deskstop where all the codepages are present and on Windows IoT Core where some codepages are not present.